flash/data/process.py

```diff
@@ -101,6 +101,151 @@ class PreprocessState:
+
+                * Output: Return the tensored image and its label.
+
+        - ``per_batch_transform``: Preforms transforms on a batch.
+            In this example, we decided not to override the hook.
+
+        - ``per_sample_transform_on_device``: Preforms transform on a sample already on a ``GPU`` or ``TPU``.
+            Example::
+
+                * Input: Receive a tensored image on device and its label.
+
+                * Action: Apply random transforms.
+
+                * Output: Return an augmented tensored image on device and its label.
+
+        - ``collate``: Converts a sequence of data samples into a batch.
+            Example::
+
+                * Input: Receive a list of augmented tensored images and their respective labels.
+
+                * Action: Collate the list of images into batch.
+
+                * Output: Return a batch of images and their labels.
+
+        - ``per_batch_transform_on_device``: Preforms transform a batch already on ``GPU`` or ``TPU``.
+            Example::
+
+                * Input: Receive a batch of images and their labels.
+
+                * Action: Apply normalization on the batch by substracting the mean
+                    and dividing by the standard deviation from Imagenet.
+
+                * Output: Return a normalized augmented batch of images and their labels.
+
+    .. note::
+
+        By default, each hook will be no-op. To customize them,
+        just override the hooks and ``Flash`` will take care of calling them at the right moment.
+
+    .. note::
+
+        The ``per_sample_transform_on_device`` and ``per_batch_transform`` are mutually exclusive
+        as it will impact performances.
+
+    To change the processing behaviour only on specific stages,
+    you can prefix all the above hooks adding ``train``, ``val``, ``test`` or ``predict``.
+
+    For example, is useful to encapsulate ``predict`` logic as labels aren't availabled at inference time.
+
+    Example::
+
+        class CustomPreprocess(Preprocess):
+
+            def predict_load_data(cls, data: Any, dataset: Optional[Any] = None) -> Mapping:
+                # logic for predict data only.
+
+    Each hook is aware of the Trainer ``running stage`` through booleans as follow.
+
+    This is useful to adapt a hook internals for a stage without duplicating code.
+
+    Example::
+
+        class CustomPreprocess(Preprocess):
+
+            def load_data(cls, data: Any, dataset: Optional[Any] = None) -> Mapping:
+
+                if self.training:
+                    # logic for train
+
+                elif self.validating:
+                    # logic from validation
+
+                elif self.testing:
+                    # logic for test
+
+                elif self.predicting:
+                    # logic for predict
+
+    .. note::
+
+        It is possible to wrap a ``Dataset`` within a :class:`~flash.data.process.Preprocess` ``load_data`` function.
```

Comment:
Also can you add the :class:Dataset link?

```suggestion
        It is possible to wrap a ``Dataset`` within a :meth:`~flash.data.process.Preprocess.load_data` function.
```

Reply:
Done !